DiabloII.Application.Tests/TestContext.cs

```diff
@@ -20,25 +22,34 @@ public sealed class TestContext : IDisposable
 
         public readonly ApplicationDbContext DbContext;
 
-        public IServiceCollection Services;
-
         public readonly TestServer TestServer;
 
+        public IServiceCollection Services;
+
         public TestContext()
         {
-            TestServer = new TestServer(CreateTheWebHostBuilder());
-
-            var httpClient = ConfigureTheHttpClient(TestServer.CreateClient());
+            var hostBuilder = CreateHostBuilder();
+            var host = hostBuilder.Start();
+            var httpClient = ConfigureTheHttpClient(host.GetTestClient());
 
             HttpService = new HttpService(httpClient);
-            DbContext = TestServer.Services.GetService<ApplicationDbContext>();
+            DbContext = host.Services.GetService<ApplicationDbContext>();
         }
 
-        private IWebHostBuilder CreateTheWebHostBuilder() => new WebHostBuilder()
-            .ConfigureServices(InitializeServices)
-            .UseEnvironment("Test")
-            .ConfigureAppConfiguration((builder => builder.AddJsonFile("appsettings.Test.json")))
-            .UseStartup(typeof(TestStartup));
+        private IHostBuilder CreateHostBuilder()
+        {
+            return new HostBuilder()
+                .UseServiceProviderFactory(new AutofacServiceProviderFactory())
+                .ConfigureWebHost(webHost =>
+                {
+                    webHost
+                        .UseTestServer()
+                        .ConfigureServices(InitializeServices)
+                        .UseEnvironment("Test")
+                        .ConfigureAppConfiguration((builder => builder.AddJsonFile("appsettings.Test.json")))
+                        .UseStartup<TestStartup>();
+                });
+        }
 
         private void InitializeServices(IServiceCollection services)
         {
```

Comment:
In order to be able to mocks dependencies and to get far better performance I should replace my test server by an owin server in memory.